plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorSmokeTest.java

```diff
@@ -0,0 +1,121 @@
+import java.util.Optional;
+
+import static io.trino.plugin.iceberg.IcebergQueryRunner.createIcebergQueryRunner;
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+public abstract class BaseIcebergConnectorSmokeTest
+        extends BaseConnectorSmokeTest
+{
+    protected final String schemaName;
+    protected final FileFormat format;
+
+    public BaseIcebergConnectorSmokeTest(String schemaName, FileFormat format)
+    {
+        this.schemaName = requireNonNull(schemaName, "schemaName is null");
+        this.format = requireNonNull(format, "format is null");
+    }
+
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        return createQueryRunner(ImmutableMap.of());
+    }
+
+    protected QueryRunner createQueryRunner(Map<String, String> connectorProperties)
+            throws Exception
+    {
+        return createIcebergQueryRunner(
+                ImmutableMap.of(),
+                ImmutableMap.<String, String>builder()
+                        .put("iceberg.file-format", format.name())
+                        .putAll(connectorProperties)
+                        .buildOrThrow(),
+                createSchemaInitializer(schemaName).build(),
+                Optional.empty());
+    }
+
+    protected SchemaInitializer.Builder createSchemaInitializer(String schemaName)
+    {
+        return SchemaInitializer.builder()
+                .withSchemaName(schemaName)
+                .withClonedTpchTables(
+                        ImmutableSet.<TpchTable<?>>builder()
+                                .add(TpchTable.NATION, TpchTable.REGION)
+                                .build());
+    }
+
+    @Override
+    protected boolean hasBehavior(TestingConnectorBehavior connectorBehavior)
+    {
+        switch (connectorBehavior) {
+            case SUPPORTS_TOPN_PUSHDOWN:
+                return false;
+
+            case SUPPORTS_CREATE_VIEW:
+                return true;
+
+            case SUPPORTS_CREATE_MATERIALIZED_VIEW:
+                return true;
+
+            case SUPPORTS_DELETE:
+                return true;
+            default:
+                return super.hasBehavior(connectorBehavior);
+        }
+    }
+
+    @Test
+    @Override
+    public void testRowLevelDelete()
+    {
+        // Deletes are covered AbstractTestIcebergConnectorTest
+        assertThatThrownBy(super::testRowLevelDelete)
+                .hasStackTraceContaining("This connector only supports delete where one or more identity-transformed partitions are deleted entirely");
+    }
+
+    @Test
+    @Override
+    public void testShowCreateTable()
+    {
+        assertThat((String) computeScalar("SHOW CREATE TABLE region"))
+                .matches("" +
+                        "CREATE TABLE iceberg." + schemaName + ".region \\(\n" +
```

Comment:
Remove the `schemaName` field as it will be used only in assertion and then it can be obtained easily with `getSession().getCatalog().orElseThrow()`

Reply:
ok